### PR TITLE
FS2 - Remove "elide empty chunks"

### PR DESCRIPTION
**Pull request by diesalbla**
The FS2 library, by construction, guarantees that the `uncons` does not give, in the result, a non-empty chunk.

### CODE REVIEW

**Review by rossabaker — COMMENTED** [review]
When did this become true?  FS2-3 or FS2-2?  Is this something that could also be done on an earlier release?

**Comment by diesalbla** [comment]
> When did this become true? FS2-3 or FS2-2? Is this something that could also be done on an earlier release?

I found about this invariant this week, but it would be interesting to know how far it extends. 

If one would like to find out, here is what has to be done: checkout the release of `fs2` being used, look for the `Output` class, which would be a subclass of `Pull` (in the 3.x branch), or a subclass of `FreeC` (on a recent 2.x). If that `Output` class is private, and the only call to its constructor includes the guard not to create it with empty chunk, as in [this line](https://github.com/typelevel/fs2/blob/17eff83fe5f0d0068e626aa9e6590eba6d62655e/core/shared/src/main/scala/fs2/Pull.scala#L241), then the `uncons` should not be giving out an empty chunk.

**Comment by rossabaker** [comment]
It appears that's true in fs2-2.5, in which case we could do the same all the way back to 0.21.

File: core/src/main/scala/org/http4s/multipart/MultipartParser.scala [hunk 1]
```diff
@@ -284,17 +284,6 @@ object MultipartParser {
     //the last boundary
     def checkIfLast(c: Chunk[Byte], rest: Stream[F, Byte]): SplitStream[F] = {
       //Elide empty chunks until nonemptychunk is found
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/multipart/MultipartParser.scala:286]: This comment can be removed.

**Review by rossabaker — APPROVED** [review]
This looks good after the comment is stripped.  I'll probably try to cherry-pick it to 0.21, unless you want to reopen with the same logic against that.

**Comment by rossabaker** [comment]
It cherry-picked cleanly to series/0.21! :tada:

**Comment by rossabaker** [comment]
Well, I should have run the tests instead of relying on reason and the compiler:

```
[error]   ! drain the epilogue with chunk size 3 (3 ms)
[error]    java.lang.RuntimeException: Chunk.empty.apply(0) (Chunk.scala:514)
[error] fs2.Chunk$EmptyChunk.apply(Chunk.scala:514)
[error] fs2.Chunk$EmptyChunk.apply(Chunk.scala:512)
[error] org.http4s.multipart.MultipartParser$.checkIfLast$1(MultipartParser.scala:312)
```

I'm going to revert the cherry-pick and merge the original.

**Pull request merged**